### PR TITLE
Feature/automatic project detection

### PR DESCRIPTION
**Pull request by hananeikhelef**
**Work Done**

- Automatic project detection feature implemented

### CODE REVIEW

File: app/src/client/injected/injected-app.tsx [hunk 1]
```diff
@@ -41,40 +40,60 @@ const Button = styled.button`
...
+      return projectName;
+    }
+  }
+
+  /**
+   * This function fetches projects names on Github and return the title of the project
+   */
+  getGcpProjectId(): string{
+    if (localStorage.getItem(this.getProjectNameFromGithub()) !==  null ) {
+      return localStorage.getItem(this.getProjectNameFromGithub());
```
Comment (jamohile) [inline app/src/client/injected/injected-app.tsx:85]: Nit: this second call could be removed by using a local variable

File: app/src/client/injected/injected-app.tsx [hunk 2]
```diff
@@ -41,40 +40,60 @@ const Button = styled.button`
...
+
+  /**
+   * This function fetches projects names on Github and return the title of the project
+   */
+  getProjectNameFromGithub(): string{
+    let title = document.querySelector(".js-path-segment:first-child");
+    if (title !== null){
+      var projectName = title.querySelector("a[data-pjax='true'] span").innerHTML;
+      return projectName;
+    }
```
Comment (jamohile) [inline app/src/client/injected/injected-app.tsx:77]: I just checked, you don't need this logic to get the title. You can call title.innerText and get the same thing!

Comment (hananeikhelef) [reply app/src/client/injected/injected-app.tsx:77]: You're right ! Done and thank you :)

File: app/src/client/injected/injected-app.tsx (hunk 1 above)
Comment (hananeikhelef) [reply app/src/client/injected/injected-app.tsx]: Done :)

File: app/src/client/injected/injected-app.tsx [hunk 3]
```diff
@@ -69,7 +69,50 @@ interface InjectedAppState{
...
+    chrome.storage.sync.get(['projects'], function(result) {
+      console.log("results are", result);
+      console.log("results are here :", result.key);
+    });
+
+    return projectId;
+  }
   
+
+
```
Comment (KushalD1999) [inline app/src/client/injected/injected-app.tsx]: NIT: Remove extra lines

File: app/src/client/injected/injected-app.tsx [hunk 4]
```diff
@@ -81,24 +81,23 @@ interface InjectedAppState{
...
-    let projectId = this.getProjectNameFromGithub();
+    let projectIdList = this.getProjectNameFromGithub();
     // Save it using the Chrome extension storage API.
-    chrome.storage.sync.set({'projects': projectId}, function() {
-      // Notify that we saved.
-      alert('Settings saved');
-    });
+    // chrome.storage.sync.set({'projects': projectIdList}, function() {
+    //   // Notify that we saved.
+    //   alert('Settings saved');
```
Comment (KushalD1999) [inline app/src/client/injected/injected-app.tsx]: Nit: Clean up

File: app/manifest.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
   "browser_action": {
     "default_popup": "src/client/popup/popup.html"
   },
-  "permissions": ["activeTab", "tabs", "identity", "storage"],
+  "permissions": ["activeTab", "tabs", "identity", "storage", "tabs"],
```
Comment (KushalD1999) [inline app/manifest.json]: Remove "Tabs". It is already present in the permissions array

File: app/src/client/injected/injected-app.tsx [hunk 6]
```diff
@@ -72,32 +72,29 @@ interface InjectedAppState{
...
-    //   // Notify that we saved.
-    //   alert('Settings saved');
-    // });
-
-    chrome.storage.sync.get(['projects'], function(result) {
-      console.log("results are", result);
-      console.log("results are");
+    chrome.storage.sync.set({'projects': projects}, function() {
+      // Notify that we saved.
+      console.log("projects are ", projects);
```
Comment (KushalD1999) [inline app/src/client/injected/injected-app.tsx]: NIT: Remove debug statement

**Pull request merged**